### PR TITLE
train: allow passing custom learning rate optimizer

### PR DESCRIPTION
**Pull request by breznak**
- allows setting customized optimizer `train(optimizer=torch.optim.MyCustomOptimizer(..))` 
- TODO add early-stopping
- TODO ideally avoid extra steps for learning-rate management (leave it to the optimizer) 

For #298

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -291,11 +297,11 @@ def train():
                 if changed:
                     cfg.delayed_settings = [x for x in cfg.delayed_settings if x[0] > iteration]
 
-                # Warm up by linearly interpolating the learning rate from some smaller value
+                # Warm up by linearly interpolating the learning rate from some smaller value 
                 if cfg.lr_warmup_until > 0 and iteration <= cfg.lr_warmup_until:
```
Comment (breznak) [inline train.py]: can we leave this fine tuning, lr management to the optimizer (or other code provided by the framework)?

**Comment by breznak** [comment]
Please feel free to take over this, I'll be out for the next week
